Xamarin.Forms.Core/ResourceDictionary.cs

```diff
@@ -34,6 +54,39 @@ public class ResourceDictionary : IResourceDictionary, IDictionary<string, objec
 		}
 
 		ResourceDictionary _mergedInstance;
+		ICollection<ResourceDictionary> _mergedDictionaries;
+		public ICollection<ResourceDictionary> MergedDictionaries { get { return _mergedDictionaries; } }
+
+		public ResourceDictionary()
+		{
+			var collection = new ObservableCollection<ResourceDictionary>();
+			collection.CollectionChanged += MergedDictionaries_CollectionChanged;
+			_mergedDictionaries = collection;
+		}
+
+		private void MergedDictionaries_CollectionChanged(object sender, System.Collections.Specialized.NotifyCollectionChangedEventArgs e)
+		{
+			// New Items
+			var newItems = e.NewItems?.Cast<ResourceDictionary>();
+			if (newItems != null)
+			{
+				foreach (var item in newItems)
+					item.ValuesChanged += Item_ValuesChanged;
+
+				if (newItems.Count() > 0)
+					OnValuesChanged(newItems.SelectMany(x => x).ToArray());
+			}
+			// Old Items
+			var oldItems = e.OldItems?.Cast<ResourceDictionary>();
+			if (oldItems != null)
+				foreach (var item in oldItems)
+					item.ValuesChanged -= Item_ValuesChanged;
```

Comment:
@StephaneDelcroix - Add/Remove/Move/Replace all work under the current code. The only issue I found is with Clear, where it won't pass through the oldItems. As such I have 3 options. Create my own class that inherits from ObservableCollection and override the Clear Method, Create my own list type from IList, or keep a secondary list, so I can iterate through them when cleared. 

None are particularly appealing, but overriding the Clear method and getting it to do an individual remove, seems like the easiest approach. While I know its a performance issue, its only a performance issue in relation to the ObservableCollections intended operation of being bound to a UI Element.

               public class MoreObservableCollection<T> : ObservableCollection<T>
		{
			protected override void ClearItems()
			{
				IList<T> items = new List<T>(this);
				foreach (T item in items)
					Remove(item);
			}
		}

Open to other ideas.